GitHgMirror.Runner/Services/GitCommandExecutor.cs

```diff
@@ -52,12 +52,34 @@ public void PushToGit(Uri gitCloneUri, string cloneDirectoryPath, string pushRef
                         // would be force push and completely overwrite the remote repo's content. This would always
                         // succeed no matter what is there but could wipe out changes made between the repo was fetched
                         // and pushed.
+                        var upToDate = false;
+
                         try
                         {
-                            repository.Network.Push(repository.Network.Remotes[remoteName], reference.CanonicalName);
+                            repository.Network.Push(
+                                repository.Network.Remotes[remoteName],
+                                reference.CanonicalName,
+                                new PushOptions
+                                {
+                                    // Here we check if is there anything to push, if not cancel the operation.
+                                    OnNegotiationCompletedBeforePush = (updates) =>

```

Comment:
You don't need parentheses around the lambda's parameter when it's just one argument.
(actually static code analysis would complain about that if this solution had it)
```suggestion
                                    OnNegotiationCompletedBeforePush = updates =>
```

Reply:
Except in JavaScript, of course 😁

Reply:
Yes of course.

BTW I like to use parentheses in this case because without them I feel something missing:
No parameters, we use them `() => ...`
One parameter, we don't `param => ...`
More than one parameter, we use them `(param1, param2, ...) => ...`

Reply:
Good point. Makes a lot of sense. But it would make the bulk of our lambda expressions quite ugly! Imagine this:

```c#
collection
    .Where((arg) => arg is {})
    .Select((param) => param.Prop)
    .Where((prop) => ...)
```

Reply:
Personally I like how airy it is without the extra parentheses. Also multiple argument lambdas are rare (I've only ever used Aggregate and the item+index overload of Select) so it's not really an aesthetic consideration for me.

At any rate, this style is already globally enforced in all C# code that uses Lombiq.Analyzers. So by now it is basically set in stone.

Reply:
Parentheses are part of the syntax, so they can't be ugly, but yes, you're both right, without them it's airy.